datastore/importers/nmdb-import-systeminfo/Parser.cpp

```diff
@@ -0,0 +1,238 @@
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+// =============================================================================
+// Maintained by Sandia National Laboratories <Netmeld@sandia.gov>
+// =============================================================================
+#include "Parser.hpp"
+
+// =============================================================================
+// Parser logic
+// =============================================================================
+Parser::Parser() : Parser::base_type(start)
+{
+    start =
+        (*qi::eol > systemInfo > -qi::eol)[(qi::_val = pnx::bind(&Parser::getData, this))];
+
+    token =
+        +qi::ascii::graph;
+
+    hostName =
+        ("Host Name: " 
+        > token[pnx::bind(&nmdo::DeviceInformation::setDeviceId, &data.devInfo, qi::_1)] 
+        > qi::eol);
+
+    osName =
+        ("OS Name: " 
+        > token[pnx::bind(&nmdo::OperatingSystem::setProductName, &data.os, qi::_1)][pnx::bind(&nmdo::OperatingSystem::setCpe, &data.os, qi::_1)] 
+        > qi::eol);
+
+    osVersion =
+        ("OS Version: " 
+        > token[pnx::bind(&nmdo::OperatingSystem::setProductVersion, &data.os, qi::_1)] 
+        > qi::eol);
+
+    osManufacturer =
+        ("OS Manufacturer: " 
+        > token[pnx::bind(&nmdo::OperatingSystem::setVendorName, &data.os, qi::_1)] 
+        > qi::eol);
+
+    osConfiguration =
+        ("OS Configuration: " 
+        > token[pnx::bind(&nmdo::DeviceInformation::setDescription, &data.devInfo, qi::_1)] 
+        > qi::eol);
+
+    systemManufacturer =
+        ("System Manufacturer: " 
+        > token[pnx::bind(&nmdo::DeviceInformation::setVendor, &data.devInfo, qi::_1)] 
+        > qi::eol);
+
+    systemModel =
+        ("System Model: " 
+        > token[pnx::bind(&nmdo::DeviceInformation::setModel, &data.devInfo, qi::_1)] 
+        > qi::eol);
+
+    systemType =
+        ("System Type: "
+        > token[pnx::bind(&nmdo::DeviceInformation::setDeviceType, &data.devInfo, qi::_1)] 
+        > qi::eol);
+
+    domain =
+        ("Domain: " 
+        > token 
+        > qi::eol); 
+
+    hotfix =
+        ("[" >> +qi::ascii::digit >> "]:" >> qi::as_string[qi::ascii::string("KB") >> +qi::ascii::digit][pnx::bind(&Parser::addHotfix, this, qi::_1)]);
+
+    hotfixes =
+        ("Hotfix(s): " 
+        >> (+qi::ascii::print - qi::eol) 
+        >> qi::eol
+        > ( *(hotfix > qi::eol) | ignoredLine));
```

Comment:
I don't think this will work as expected, it probably should be one or more (i.e., `+`).  Saying zero or more means the logic is a success even when the rule fails to parse anything; so it doesn't actually ever consume the bad line and the following rules will fail.